torchao/experimental/ops/mps/register.mm

```diff
@@ -82,6 +94,31 @@ Tensor linear_mps_kernel(
   return C;
 }
 
+template <int nbit>
+Tensor linear_mps_kernel_meta(
+    const Tensor& A,
+    const Tensor& B,
+    int64_t group_size,
+    const Tensor& SZ) {
+  TORCH_CHECK(
+      A.is_meta(), __func__, "A is on ", A.device(), " but expected on meta");
+  TORCH_CHECK(
+      B.is_meta(), __func__, "B is on ", B.device(), " but expected on meta");
+  TORCH_CHECK(
+      SZ.is_meta(),
+      __func__,
+      "SZ is on ",
+      SZ.device(),
+      " but expected on meta");
+
+  check_linear_mps_args<nbit>(A, B, group_size, SZ);
+
+  auto M = A.size(0);
+  auto N = B.size(0);
+
+  return at::empty({M, N}, A.options()).to("meta");
+}
+
 // LowBit Packing on CPU Backend
```

Comment:
Should we add headers to the packed weights?  Do we want mps/metal in the name of the packing op?

Reply:
Headers to the packed weights? What do you mean by that?

Reply:
We have a header to add to packed weights that identifies the packing format: https://fburl.com/gmws8i73.  When the kernel reads the packed weights, it can check the header to make sure the packing format is as expected.

It is added to both the CPU linear op (https://fburl.com/90ok50q9) and the CPU embedding op (https://fburl.com/ydhdq5ye).

I'm wondering if the packed weights for MPS should also have the header.  I guess if we decide to do that, it can come in a follow-up PR because this PR doesn't pack the weights.  cc @kimishpatel for his thoughts.

Reply:
@kimishpatel I don't follow this:

> AFAIU, for metal kernels to have different packed weights

You can have a header even if you only have one current format, no?